### PR TITLE
Improve typings and docs for XrPlane

### PR DESCRIPTION
**Pull request by willeastcott**
Also enhances the API docs.

I confirm I have read the [contributing guidelines](https://github.com/playcanvas/engine/blob/master/.github/CONTRIBUTING.md) and signed the [Contributor License Agreement](https://docs.google.com/a/playcanvas.com/forms/d/1Ih69zQfJG-QDLIEpHr6CsaAs6fPORNOVnMv5nuo0cjk/viewform).

### CODE REVIEW

File: src/framework/xr/xr-plane.js [hunk 1]
```diff
@@ -159,10 +161,12 @@ class XrPlane extends EventHandler {
     }
 
     /**
-     * Array of DOMPointReadOnly objects. DOMPointReadOnly is an object with `x y z` properties
-     * that defines a local point of a plane's polygon.
+     * Gets the array of points that define the polygon of the plane in its local coordinate space.
```
Comment (LeXXik) [inline src/framework/xr/xr-plane.js:164]: "an array". I believe the very first occurrence of a thing uses "a", and "the" afterwards.

**Review by LeXXik — APPROVED** [review]

File: src/framework/xr/xr-plane.js (hunk 1 above)
Comment (willeastcott) [reply src/framework/xr/xr-plane.js:174]: The points are not created in response to calling the getter. The array lives on the WebXR XRPlane and the getter simply returns them.

**Review by LeXXik — APPROVED** [review]

File: src/framework/xr/xr-plane.js [hunk 2]
```diff
@@ -150,24 +152,36 @@ class XrPlane extends EventHandler {
     }
 
     /**
-     * Plane's specific orientation (horizontal or vertical) or null if orientation is anything else.
+     * Gets the plane's specific orientation. This can be "horizontal" for planes that are parallel
+     * to the ground, "vertical" for planes that are perpendicular to the ground, or `null` if the
+     * orientation is different or unknown.
      *
      * @type {string|null}
+     * @example
```
Comment (mvaligursky) [inline src/framework/xr/xr-plane.js:160]: do we want to list the possible strings in the `@type` tag instead of `string`?

**Review by mvaligursky — APPROVED** [review]

File: src/framework/xr/xr-plane.js (hunk 2 above)
Comment (willeastcott) [reply src/framework/xr/xr-plane.js:160]: Done.

**Pull request merged**